### PR TITLE
Clickable clue references

### PR DESCRIPTION
**Pull request by fishy15**
Allows user to click on clue references and go to the matching clue. Implements #304.

Would definitely appreciate feedback on UI, I simply just converted it into a button to make it stand out, but something else could be done as well.

### CODE REVIEW

File: src/components/Chat/Chat.js [hunk 1]
```diff
@@ -310,7 +311,7 @@ export default class Chat extends Component {
             {token.type === 'emoji' ? (
               <Emoji emoji={token.data} big={bigEmoji} />
             ) : token.type === 'clueref' ? (
-              token.data // for now, don't do anything special to cluerefs
+              this.renderClueRef(token.data)
```
Comment (stevenhao) [inline src/components/Chat/Chat.js:314]: interesting that this code already existed! i remember wanting this feature before but i guess never pushed it through

**Comment by fishy15** [comment]
I believe the logic works right now, you can review that if you want. But if you have any design ideas for the component, you can implement those. Currently I am thinking of making it look like a hyperlink.

**Comment by stevenhao** [comment]
image
styling seems alright, i think the next step would be to make the feature more discoverable. any ideas?

**Comment by stevenhao** [comment]
merging this now

**Pull request merged**